compiler/lexer.cpp

```diff
@@ -1379,9 +1381,22 @@ substallpatterns(unsigned char* line, int buffersize)
     unsigned char *start, *end;
     int prefixlen;
 
-    int depth = 0;
+    std::unordered_set<std::string> blocked_macros;
+
+    auto enter_macro = [&](const char* start, size_t prefixlen, macro_t* out) -> bool {
+        if (!find_subst(start, prefixlen, out))
+            return false;
+        if (blocked_macros.count(out->first)) {
+            printf("%s\n", std::string((const char *)start, prefixlen).c_str());

```

Comment:
Left this debug print in.

Also, this PR is against the other PR branch and not master.

Reply:
Thanks, fixed.